docs/ug/reportConfig.md

```diff
@@ -225,6 +228,29 @@ For each repository in the `repos` list, you can specify:
           * Default: The default branch of the repository.
         * `blurb`: A short description of the branch that appears in the report.
             * For more detailed descriptions, you can use a separate `blurbs.md` file.
+        * `since`: Start date of commits to include in the analysis. Accepted formats:
+          * Date only
+            * `dd/MM/yyyy`
+          * Date and time
+            * `dd/MM/yyyy HH:mm` or
+            * `dd/MM/yyyy HH:mm:ss`
+
+        * `until`: End date of commits to include in the analysis. Accepted formats:
+          * **same as the above**
+
+<box type="info" seamless>
+Behavior When Both Report Config Dates & CLI Flags Are Provided:
+
+- If `since` date is specified in both `report-config.yaml` and CLI, the **Config dates take precedence** for that repository. The same applies to `until` date.
+- The Config date range must fall within the **CLI date boundaries**. If it exceeds, the program exits with an error.
+    - Example:
+        - CLI Date - Since: 10/10/2024, Until: 20/10/2024
+        - Valid Config Date - Since: 11/10/2024, Until: 19/10/2024
+        - Invalid Config Date - Since: 9/10/2024, Until: 21/10/2024
+- If only one of the CLI flags is provided, the Config dates will not be restricted by the missing CLI flag.
+- If Config dates are missing, the CLI flags are used.
+- If neither Config nor CLI provides dates, default date ranges are used.
```

Comment:
```suggestion
**Behavior of `since`/`until` in report-config.yaml vs. `--since`/`--until` in CLI**

* Config dates take precedence over CLI flags when both are provided.
* Config date range must fall *within* the CLI date range; otherwise the program exits with an error.
    * Example with CLI Since=10/10/2024, Until=20/10/2024:
        * Valid: 11/10/2024–19/10/2024
        * Invalid: 9/10/2024–21/10/2024
* If only one CLI flag is provided, the missing CLI value does not restrict config dates.
* If config dates are missing, CLI values are used.
* If neither provides dates, default ranges are applied.
```

This should be clearer.